DirectXShaderCompiler vendored in SDL_shadercross fails to build with Xcode 26.4 / Apple clang 21 on macOS

Hi,
I am hitting a build failure in the vendored external/DirectXShaderCompiler when building SDL_shadercross on macOS with the current Xcode. I have replaced the with the current head from microsoft/DirectXShaderCompiler but that did not fix the issue.

Environment:
- macOS on Apple Silicon
- Xcode 26.4
- Apple clang 21.0.0
- SDK: MacOSX26.4.sdk
- SDL_shadercross [3a31448](https://github.com/libsdl-org/SDL_shadercross/commit/3a314481ede01f5d966f0e834019942e34c02b82)
- DirectXShaderCompiler [67a8e24](https://github.com/microsoft/DirectXShaderCompiler/commit/67a8e24c8dd26c3df3e8389cbdf412690dab57e6)

I have SDL_shadercross vendored in my game engine. The build fails while compiling vendor/SDL_shadercross/external/DirectXShaderCompiler/... sources, with errors coming from include/llvm/ADT/StringRef.h.
Representative error:
```
/Users/frank/Projects/bms/adventurer/vendor/SDL_shadercross/external/DirectXShaderCompiler/include/llvm/ADT/StringRef.h:579:10: error: 'is_nothrow_constructible' cannot be specialized: Users are not allowed to specialize this standard library entity [-Winvalid-specialization]
  579 |   struct is_nothrow_constructible <std::string, llvm::StringRef>
      |          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX26.4.sdk/usr/include/c++/v1/__type_traits/is_nothrow_constructible.h:23:8: note: marked '_Clang::no_specializations' here
   23 | struct _LIBCPP_NO_SPECIALIZATIONS is_nothrow_constructible
      | 
```
I see the same error for:
- is_nothrow_constructible<std::string, llvm::StringRef>
- is_nothrow_constructible<std::string, llvm::StringRef &>
- is_nothrow_constructible<std::string, const llvm::StringRef &>
This appears to be caused by newer libc++ in Xcode 26.4 rejecting specialization of this standard library trait.

Full log: https://pastebin.com/XKX67V5N

--- a/include/llvm/ADT/StringRef.h
+++ b/include/llvm/ADT/StringRef.h
@@ -571,9 +571,12 @@ namespace llvm {
 }
 
 // HLSL Change Starts
-// StringRef provides an operator string; that trips up the std::pair noexcept specification,
-// which (a) enables the moves constructor (because conversion is allowed), but (b)
-// misclassifies the the construction as nothrow.
+// StringRef provides an operator string; that trips up the std::pair noexcept
+// specification, which (a) enables the moves constructor (because conversion is
+// allowed), but (b) misclassifies the the construction as nothrow. Newer libc++
+// releases reject user specializations of this trait outright, and also compute
+// the trait correctly without help.
+#if !defined(_LIBCPP_VERSION)
 namespace std {
   template<>
   struct is_nothrow_constructible <std::string, llvm::StringRef>
@@ -588,6 +591,7 @@ namespace std {
     : std::false_type {
   };
 }
+#endif
 // HLSL Change Ends
 
 #endif

